### PR TITLE
WIP: Add plasma-phone package

### PR DESCRIPTION
**Pull request by Slabity**
###### Motivation for this change
This is a work in progress for getting plasma-mobile software working in NixOS.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/plasma-5/srcs.nix
Comment (samueldr) [inline pkgs/desktops/plasma-5/srcs.nix]: (Nothing to do right now, but to be done eventually before merging.)

Their `invent` platform is a GitLab server, `fetchFromGitLab` should be used to fetch their tarballs since revision tarballs, from most repo hosting systems, might eventually change as they are generated, making the sha256 unstable.

**Comment by Slabity** [comment]
I'm currently trying to get a configuration option to enable the mobile shell. I can get it working manually by running `plasmashell -p org.kde.plasma.phone` (after killing the default plasmashell), but I'd like to try and match it to the current plasma-desktop as closely as possible.

If anyone has a better idea on how to do this it would be greatly appreciated.

File: pkgs/desktops/plasma-5/srcs.nix
Comment (Slabity) [reply pkgs/desktops/plasma-5/srcs.nix]: What would be the process of getting this on a mirror to match all the other packages in `srcs.nix`?

Comment (samueldr) [reply pkgs/desktops/plasma-5/srcs.nix]: This bit

 * https://github.com/NixOS/nixpkgs/blob/803e956919ef1c14a0403250201d510c36ace64e/maintainers/scripts/fetch-kde-qt.sh#L6-L13

Ends up using either of:

 * https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/kde/fetch.sh
 * https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/plasma-5/fetch.sh

Depending of where exactly it would go (among other `srcs.nix` handled by that script). So basically whenever kde/plasma publishes a released source for that.

**Comment by aanderse** [comment]
ping (triage)
status?

**Review by Thra11 — CHANGES_REQUESTED** [review]

**Comment by Thra11** [comment]
There a few other plasma-mobile packages in stable srcs.nix now too.

**Comment by Slabity** [comment]
Sorry to disappoint, but I'm not really able to work on this further. I will be closing this as I believe there is a much more organized 'Mobile NixOS' initiative going on.

**Comment by samueldr** [comment]
Anyone following this PR, or finding this PR, and is concerned about this being dropped. I'm packaging it up.

I have a WIP dirty branch here:

 - https://github.com/samueldr/nixpkgs/tree/wip/plamo

Already in it is a couple PRs, and the changes required for Plasma Mobile to work. Though things don't *exactly* work out of the box. At least for the DM session. So stay tuned for a follow-up PR!

**Comment by samueldr** [comment]
So for anyone *still* following this PR, to help push myself into finishing this, #121345 was opened. Not really ready for reviews, but something to look at.

**Pull request closed without merging**
